### PR TITLE
Use explicit offset to set child consensus times during finalization

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Avoids skipped nanos in consensus times by using an independent `nextNanoOffset` instead of the index of the pending record in `childRecords`.

### CODE REVIEW

File: test-clients/src/main/resource/contract/solidity/hip-206/HederaTokenService.sol [hunk 1]
```diff
@@ -386,16 +386,6 @@ abstract contract HederaTokenService is HederaResponseCodes {
         (responseCode) = success ? abi.decode(result, (int32)) : HederaResponseCodes.UNKNOWN;
     }
 
-    /// Operation to unpause token
-    /// @param token The token address to be unpaused
-    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
-    function unpauseToken(address token) external returns (int responseCode)
```
Comment (Neeharika-Sompalli) [inline test-clients/src/main/resource/contract/solidity/hip-206/HederaTokenService.sol:392]: Is this removed intentionally ?

File: test-clients/src/main/resource/contract/solidity/hip-206/IHederaTokenService.sol [hunk 2]
```diff
@@ -613,14 +613,6 @@ interface IHederaTokenService {
         external
         returns (int64 responseCode, bool defaultKycStatus);
 
-    /// Query token expiry info
```
Comment (Neeharika-Sompalli) [inline test-clients/src/main/resource/contract/solidity/hip-206/IHederaTokenService.sol:616]: same here

File: test-clients/src/main/resource/contract/solidity/hip-206/HederaTokenService.sol (hunk 1 above)
Comment (tinker-michaelj) [reply test-clients/src/main/resource/contract/solidity/hip-206/HederaTokenService.sol:392]: Yep; a proper local Solidity dev setup wouldn't be using these files in any form, so I just needed to delete this to compile that test case.

File: test-clients/src/main/resource/contract/solidity/hip-206/IHederaTokenService.sol (hunk 2 above)
Comment (tinker-michaelj) [reply test-clients/src/main/resource/contract/solidity/hip-206/IHederaTokenService.sol:616]: Also intentional!

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java [hunk 3]
```diff
@@ -93,7 +97,39 @@ public List<HapiSpec> getSpecsInSuite() {
                 eip1014AddressAlwaysHasPriority(),
                 immutableContractKeyFormIsStandard(),
                 updateAutoRenewAccountWorks(),
-                updateStakingFieldsWorks());
+                updateStakingFieldsWorks(),
+                consTimeManagementWorksWithRevertedInternalCreations());
+    }
+
+    private HapiSpec consTimeManagementWorksWithRevertedInternalCreations() {
```
Comment (Neeharika-Sompalli) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java:104]: nit: We could probably move this to ContractCallSuite

Comment (tinker-michaelj) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java]: Done.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by mhess-swl — APPROVED** [review]

**Pull request merged**